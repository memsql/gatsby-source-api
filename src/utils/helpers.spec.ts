import {
    getPrettyName,
    isPrimitive,
    PLUGIN_NAME,
    PRETTY_PLUGIN_NAME,
} from "utils/helpers";

describe("helper constants", () => {
    describe("PLUGIN_NAME", () => {
        it("matches the snapshot", () => {
            expect(PLUGIN_NAME).toMatchSnapshot();
        });

        it("matches its pretty snapshot", () => {
            expect(PRETTY_PLUGIN_NAME).toMatchSnapshot();
        });

        it("creates a pretty plugin instance name", () => {
            expect(getPrettyName("test")).toMatchSnapshot();
        });
    });
});

describe("isPrimitive helper", () => {
    it("determines that `string` is primitive", () => {
        expect(isPrimitive("hello")).toBe(true);
    });

    it("determines that `number` is primitive", () => {
        expect(isPrimitive(10)).toBe(true);
    });

    it("determines that `boolean` is primitive", () => {
        expect(isPrimitive(false)).toBe(true);
    });

    it("determines that `Date` is primitive", () => {
        expect(isPrimitive(new Date())).toBe(true);
    });

    it("determines that `array` is not primitive", () => {
        expect(isPrimitive([])).toBe(false);
    });

    it("determines that `object` is not primitive", () => {
        expect(isPrimitive({})).toBe(false);
    });
});
