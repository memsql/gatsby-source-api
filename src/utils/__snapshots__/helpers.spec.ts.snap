// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helper constants PLUGIN_NAME creates a pretty plugin instance name 1`] = `"[36mgatsby-source-api[39m [33mtest[39m"`;

exports[`helper constants PLUGIN_NAME matches its pretty snapshot 1`] = `"[36mgatsby-source-api[39m"`;

exports[`helper constants PLUGIN_NAME matches the snapshot 1`] = `"gatsby-source-api"`;
